### PR TITLE
Feature/copy cluster content to local cache

### PR DESCRIPTION
**Pull request by KostiaLevchenko**

### CODE REVIEW

File: src/glusterfs_lib/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,7 @@ add_library(
         xglfs_ftruncate.c
         xglfs_utimens.c
         xglfs_statfs.c
+        xglfs_readdir.c
```
Comment (kopachevsky) [inline src/glusterfs_lib/CMakeLists.txt:28]: how come we didn't have it before?

File: src/kfs_lib/kfs_init.c [hunk 2]
```diff
@@ -7,6 +7,7 @@ void *kfs_init(struct fuse_conn_info *connection) {
     if (XGLFS_STATE->gluster_api) {
         log_info("   gluster sync enabled");
         res = xglfs_init(connection);
+        read_cluster();
```
Comment (kopachevsky) [inline src/kfs_lib/kfs_init.c]: if we are not able to donwload content we should exit

**Review by kopachevsky — CHANGES_REQUESTED** [review]

File: src/kfs_lib/kfs_common.c [hunk 3]
```diff
@@ -17,6 +17,45 @@ char *str_concat(const char *s1, const char *s2) {
...
+}
+
+void copy_content(char *path) {
+    char fpath[PATH_MAX];
+    fullpath(fpath, path);
+    int in_fd, out_fd = 0, dir_fd;
+    struct fuse_file_info fi = read_struct(O_RDONLY);
+    struct fuse_file_info opendir = read_struct(O_DIRECTORY);
+    if ((in_fd = xglfs_open(path, &fi)) == -1) {
+        log_debugf("Cannot open %d\n", in_fd);
```
Comment (kopachevsky) [inline src/kfs_lib/kfs_common.c]: add infor about what path you  can't open

File: src/kfs_lib/kfs_init.c (hunk 2 above)
Comment (KostiaLevchenko) [reply src/kfs_lib/kfs_init.c]: fixed 9566f3f

File: src/glusterfs_lib/CMakeLists.txt (hunk 1 above)
Comment (KostiaLevchenko) [reply src/glusterfs_lib/CMakeLists.txt:28]: reading functions of glusterfs wasn't necessary before

File: src/kfs_lib/includes/kfs_common.h [hunk 4]
```diff
@@ -19,9 +19,13 @@
 char *MOUNT_PATH;
 char *LOCAL_DISC_CACHE_PATH;
 char *LOG_CONFIG_PATH;
+#define COPYMODE        0644
```
Comment (kopachevsky) [inline src/kfs_lib/includes/kfs_common.h:22]: macros make sense when you change it to different builds, don't put it everywhere, use constants instead

**Review by kopachevsky — APPROVED** [review]

**Pull request merged**